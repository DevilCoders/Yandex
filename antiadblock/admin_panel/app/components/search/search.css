.search
{
    width: 100%;
    padding: 20px;
    box-sizing: border-box;

    overflow: auto;

    &__wrapper
    {
        position: relative;
        margin-top: 15px;
    }

    &__active-before,
    &__active-after
    {
        display: inline-block;
        height: 30px;
        text-align: right;
    }

    &__active
    {
        display: inline-block;
    }

    &__active,
    &__active-after
    {
        margin-left: 10px;
    }

    &__count
    {
        height: 30px;
    }

    &__hint
    {
        font-weight: 500;
        display: inline-block;
    }
}
