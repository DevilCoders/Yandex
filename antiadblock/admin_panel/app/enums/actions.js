export const ACTIONS = {
    // Временно скрыт мониторинг на сервисах, т.к. он пока не работает
    // GRAPHS: 'graphs',
    CONFIGS: 'configs',
    AUDIT: 'audit',
    INFO: 'info',
    UTILS: 'utils',
    // ANTIADB-2636
    // HEALTH: 'health',
    RULES: 'rules',
    SBS_RESULT_CHECKS: 'sbs-result-checks'
};

export const SECRET_ACTIONS = {
    SCREENSHOTS: 'screenshots-checks',
    PROFILES: 'sbs-profiles',
    LOGS: 'logs'
};
