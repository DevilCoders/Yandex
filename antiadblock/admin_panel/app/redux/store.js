import {createStore, compose, applyMiddleware} from 'redux';
import reducer from 'app/reducers/index';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';

let composeEnhancers = compose;

export function configureStore(state = {}) {
    if (typeof window !== 'undefined' && window.__REDUX_DEVTOOLS_EXTENSION__) {
        composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__;
    }
    if (process.env.NODE_ENV === 'local') {
        return createStore(reducer, state, composeEnhancers(applyMiddleware(thunkMiddleware, createLogger())));
    }
    return createStore(reducer, state, composeEnhancers(applyMiddleware(thunkMiddleware)));
}
