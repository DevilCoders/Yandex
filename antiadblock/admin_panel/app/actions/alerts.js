import servicesApi from 'app/api/services';
import {setErrors, clearErrors} from 'app/actions/errors';
import {getServices, getErrors} from 'app/reducers/index';
import {getById} from 'app/reducers/errors';
import {timeIsOver} from 'app/lib/errors-logic';

export const START_ALERTS_LOADING = 'START_ALERTS_LOADING';
export const END_ALERTS_LOADING = 'END_ALERTS_LOADING';

export function fetchAlerts() {
    return (dispatch, getState) => {
        dispatch(startAlertsLoading());

        return servicesApi.fetchAlerts().then(alerts => {
            const errors = getById(getErrors(getState()), 'dashboard');
            if (errors && errors.length) {
                dispatch(clearErrors('dashboard'));
            }

            return dispatch(endAlertsLoading(alerts.items));
        }, error => {
            const lastUpdateTime = getServices(getState()).lastUpdateTimeAlerts;

            if (timeIsOver(lastUpdateTime)) {
                return dispatch(setErrors('dashboard', [error.message]));
            }
        });
    };
}

export function startAlertsLoading() {
    return {
        type: START_ALERTS_LOADING
    };
}

export function endAlertsLoading(items) {
    return {
        type: END_ALERTS_LOADING,
        items
    };
}
