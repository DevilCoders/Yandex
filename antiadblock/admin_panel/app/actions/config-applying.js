import configApi from 'app/api/config';
import {setGlobalErrors} from 'app/actions/errors';

export const OPEN_CONFIG_APPLYING = 'OPEN_CONFIG_APPLYING';
export const CLOSE_CONFIG_APPLYING = 'CLOSE_CONFIG_APPLYING';
export const START_ACTIVE_CONFIG_LOADING = 'START_ACTIVE_CONFIG_LOADING';
export const END_ACTIVE_CONFIG_LOADING = 'END_ACTIVE_CONFIG_LOADING';

export function openConfigApplying(serviceId, id, configData, target) {
    return {
        type: OPEN_CONFIG_APPLYING,
        serviceId,
        id,
        configData,
        target
    };
}

export function closeConfigApplying(id) {
    return {
        type: CLOSE_CONFIG_APPLYING,
        id
    };
}

export function fetchCurrentConfig(labelId, target) {
    return dispatch => {
        dispatch(startActiveConfigLoading(labelId, target));

        return configApi.fetchCurrentConfig(labelId, target).then(config => {
            return dispatch(endActiveConfigLoading(labelId, target, config));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startActiveConfigLoading(labelId, target) {
    return {
        type: START_ACTIVE_CONFIG_LOADING,
        labelId,
        target
    };
}

export function endActiveConfigLoading(labelId, target, config) {
    return {
        type: END_ACTIVE_CONFIG_LOADING,
        labelId,
        target,
        config
    };
}
