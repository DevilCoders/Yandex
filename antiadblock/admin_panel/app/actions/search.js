import servicesApi from 'app/api/services';
import {setGlobalErrors} from 'app/actions/errors';

export const START_SEARCH = 'START_SEARCH';
export const END_SEARCH = 'END_SEARCH';

export function searchConfigs(pattern, offset, limit, active) {
    return dispatch => {
        dispatch(startSearchConfigs(pattern, offset, limit, active));

        return servicesApi.searchConfigs(pattern, offset, limit, active).then(configs => {
            return dispatch(endSearchConfigs(pattern, configs));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startSearchConfigs(pattern, offset, limit, active) {
    return {
        type: START_SEARCH,
        pattern,
        offset,
        limit,
        active
    };
}

export function endSearchConfigs(pattern, configs) {
    return {
        type: END_SEARCH,
        pattern,
        configs
    };
}
