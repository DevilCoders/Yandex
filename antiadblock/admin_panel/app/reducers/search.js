import * as searchActions from 'app/actions/search';
import update from 'app/lib/update';

export default function(state, action) {
    const {type} = action;

    const initialState = {
        byId: {},
        items: [],
        loaded: false,
        total: 0
    };

    state = state || initialState;

    switch (type) {
        case searchActions.START_SEARCH: {
            return update(state, {
                loaded: {
                    $set: action.offset > 0
                },
                items: {
                    $set: action.offset > 0 ? state.items : []
                },
                pattern: {
                    $set: action.pattern
                },
                active: {
                    $set: action.active
                }
            });
        }
        case searchActions.END_SEARCH: {
            return update(state, {
                items: {
                    $push: action.configs.items.map(config => config.id)
                },
                byId: {
                    $merge: action.configs.items.reduce((map, config) => {
                        map[config.id] = config;

                        return map;
                    }, {})
                },
                pattern: {
                    $set: action.pattern
                },
                offset: {
                    $set: state.offset + action.configs.items.length
                },
                total: {
                    $set: action.configs.total
                },
                loaded: {
                    $set: true
                }
            });
        }
        default: {
            return state;
        }
    }
}
