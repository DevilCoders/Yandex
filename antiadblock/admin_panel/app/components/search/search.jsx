import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import queryString from 'query-string';

import {searchConfigs} from 'app/actions/search';
import {getSearch} from 'app/reducers/index';

import Tumbler from 'app/components/tumbler/tumbler';

import Preloader from 'app/components/preloader/preloader';
import Bem from 'app/components/bem/bem';
import InfiniteList from 'app/components/infinite-list/infinite-list';
import SearchConfigsItem from './__item/search__item';

import ConfigDiffModal from 'app/components/config-diff-modal/config-diff-modal';
import ConfigPreview from 'app/components/config-preview/config-preview';

import Hint from 'app/components/hint/hint';

import i18n from 'app/lib/i18n';

import './search.css';
import {antiadbUrl} from 'app/lib/url';

class SearchPage extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            pattern: this.props.pattern,
            active: this.props.tumblerState
        };
        this.onChange = this.onChange.bind(this);
    }

    onChange() {
        this.setState(state => ({
            active: !state.active
        }));
        this.context.router.history.push(antiadbUrl(`/search?pattern=${this.state.pattern.replace(/\+/g, '%2B')}&active=${!this.state.active}`));
    }

    render() {
        return (
            <Bem
                block='search'>
                {!this.props.loaded ?
                    <Preloader key='preloader' /> :
                    ''}
                <Bem
                    block='search'
                    elem='count'>
                    {i18n('search-page', 'total')}: {this.props.total}
                </Bem>
                <Bem
                    block='search'
                    elem='active-before'
                    key='active-before'>
                    {i18n('search-page', 'active')}
                </Bem>
                <Bem
                    block='search'
                    elem='hint'>
                    <Hint
                        text={i18n('search-page', 'active-hint')}
                        to='right' />
                </Bem>
                <Bem
                    block='search'
                    elem='active'
                    key='active'>
                    <Tumbler
                        theme='normal'
                        view='default'
                        tone='grey'
                        size='s'
                        onVal='onVal'
                        offVal='offVal'
                        checked={this.state.active}
                        onChange={this.onChange} />
                </Bem>
                <Bem
                    key='wrapper'
                    block='search'
                    elem='wrapper'>
                    <InfiniteList
                        key='list'
                        wrapperMix={{
                            block: 'search',
                            elem: 'scroll-wrapper'
                        }}
                        filters={{
                            active: this.state.active
                        }}
                        requestMore={this.props.searchConfigs}
                        total={this.props.total}
                        items={this.props.configs}
                        Item={SearchConfigsItem} />
                </Bem>
                <ConfigPreview
                    key='preview' />
                <ConfigDiffModal
                    key='diff' />
            </Bem>
        );
    }
}

SearchPage.propTypes = {
    configs: PropTypes.array,
    loaded: PropTypes.bool,
    total: PropTypes.number,
    tumblerState: PropTypes.bool,
    pattern: PropTypes.string,
    searchConfigs: PropTypes.func
};

SearchPage.contextTypes = {
    router: {
        history: {
            push: PropTypes.func.isRequired
        }
    }
};

export default connect((state, props) => {
    const parsed = queryString.parse(props.location.search);
    const configs = getSearch(state);
    return {
        configs: configs.items,
        loaded: configs.loaded,
        total: configs.total,
        pattern: parsed.pattern,
        tumblerState: parsed.active === 'true'
    };
}, (dispatch, props) => {
    return {
        searchConfigs: (offset, limit, filters) => {
            const parsed = queryString.parse(props.location.search);
            return dispatch(searchConfigs(parsed.pattern, offset, limit, filters.active));
        }
    };
})(SearchPage);
