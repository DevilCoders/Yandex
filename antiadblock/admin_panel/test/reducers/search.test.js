import reducer from 'app/reducers/search';
import * as searchActionsFixtures from '../fixtures/actions/search';
import {BEFORE_SEARCH, START_SEARCH, END_SEARCH} from '../fixtures/store';

describe('search reducer', () => {
    it('should return the initial state', () => {
        expect(reducer(BEFORE_SEARCH, {})).toMatchSnapshot();
    });

    it('should handle START_SEARCH', () => {
        expect(
            reducer(START_SEARCH, searchActionsFixtures.START_SEARCH)
        ).toMatchSnapshot();
    });

    it('should handle END_SEARCH', () => {
        expect(
            reducer(END_SEARCH, searchActionsFixtures.END_SEARCH)
        ).toMatchSnapshot();
    });
});
