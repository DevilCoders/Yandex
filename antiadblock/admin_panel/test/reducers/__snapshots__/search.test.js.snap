// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search reducer should handle END_SEARCH 1`] = `
Object {
  "byId": Object {},
  "items": Array [],
  "loaded": true,
  "offset": 0,
  "pattern": "test",
  "total": 0,
  "type": "END_SEARCH",
}
`;

exports[`search reducer should handle START_SEARCH 1`] = `
Object {
  "active": true,
  "items": Array [],
  "loaded": false,
  "pattern": "test",
  "type": "START_SEARCH",
}
`;

exports[`search reducer should return the initial state 1`] = `
Object {
  "byId": Object {},
  "items": Array [],
  "loaded": false,
  "total": 0,
}
`;
